carpics_360v0.1.js

```diff
@@ -381,15 +384,144 @@ var CarPicsSpinnerAPI = (function() {
+                        thisObj.spinStatus = false;
+                        thisObj.turnStatus = false;
+                    } else {
+                        thisObj.spinStatus = thisObj.spinDefault;
+                        thisObj.turnStatus = false;
+                    }
+                }
+            })(this));
+
+            document.getElementById("zoom_in_button").addEventListener("mouseleave", (function(thisObj) {
+                return function(baseEvent) {
+                    if (thisObj.zoomed == true) {
+                        document.getElementById("zoom_in_button").style.background="#888";
+                        document.getElementById("zoom_in_button").style.color="#fff";
+                    } else {
+                        document.getElementById("zoom_in_button").style.background="#fff";
+                        document.getElementById("zoom_in_button").style.color="#333";
+                    }
+                }
+            })(this));
+
+            /*
+            * Clicking on zoomout button triggers desktop zoom out event.
+            */
+            document.getElementById("zoom_out_button").addEventListener("click", (function(thisObj) {
+                return function(baseEvent) {
+                    baseEvent.stopPropagation();
+                    CarPicsGoogleAnalytics('send', 'pageview', {'dimension1':'Click'});
+                    baseEvent.preventDefault();
+                    if (thisObj.zoomed === true) {
+                        var releaseMouse = thisObj.zoomToggle(baseEvent, true);
+                        thisObj.spinStatus = false;
+                        thisObj.turnStatus = false;
+                    } else {
+                        thisObj.spinStatus = thisObj.spinDefault;
+                        thisObj.turnStatus = false;
+                    }
+
+                }
+            })(this));
+
+            /*
+            * Clicking on hotspot button triggers display/hide hotspots
+            */
+            document.getElementById("hotspot_button").addEventListener("click", (function(thisObj) {
+                return function(baseEvent) {
+                    baseEvent.stopPropagation();
+                    CarPicsGoogleAnalytics('send', 'pageview', {'dimension1':'Click'});
+                    baseEvent.preventDefault();
+                    if (thisObj.displayHotspots == true) {
+                        // hide hotspots
+                        var hotspots = document.getElementsByClassName("hotspot");
+                        while (hotspots.length > 0) {
+                            hotspots[0].parentElement.removeChild(hotspots[0]);
+                        }
+                        thisObj.displayHotspots = false;
+                        document.getElementById("hotspot_button").innerHTML="&#8709";
+                    } else {
+                        // display hotspots
+                        for (j=0; j<thisObj.AllImages.length; j++) {
```

Comment:
Var y scary thing to put here.  Always use var when setting a variable like this.  Since this is presumably not throwing an error, this means it is pulling the j from a further up scope.  This can lead to very hard to trace down bugs.  Always be sure to define all local variables locally.